pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -0,0 +1,561 @@
+				return cluster, nil
+			}
+
+			logrus.Infof("generating service account token for cluster [%s]", cluster.Name)
+			sess, err := e.startAWSSession(cluster.Spec.EKSConfig.AmazonCredentialSecret)
+			if err != nil {
+				return cluster, nil
+			}
+
+			endpoint := string(caSecret.Data["endpoint"])
+			ca := string(caSecret.Data["ca"])
+			saToken, err := generateSAToken(sess, cluster.Spec.EKSConfig.DisplayName, endpoint, ca)
+			if err != nil {
+				return cluster, err
+			}
+
+			cluster = cluster.DeepCopy()
+			cluster.Status.APIEndpoint = endpoint
+			cluster.Status.CACert = ca
+			cluster.Status.ServiceAccountToken = saToken
+			return e.clusterClient.Update(cluster)
+		}
+
+		// check for unauthorized error
+		if failureMessage, _ := status["failureMessage"].(string); failureMessage != "" {
+			if strings.Contains(failureMessage, "403") {
+				if !apimgmtv3.ClusterConditionUpdated.IsFalse(cluster) {
+					cluster = cluster.DeepCopy()
+					apimgmtv3.ClusterConditionUpdated.False(cluster)
+					apimgmtv3.ClusterConditionUpdated.Message(cluster, "cannot access EKS, check cloud credential")
+					return e.clusterClient.Update(cluster)
+				}
+			}
+		}
+
+		if !apimgmtv3.ClusterConditionUpdated.IsTrue(cluster) {
+			cluster = cluster.DeepCopy()
+			apimgmtv3.ClusterConditionUpdated.True(cluster)
+			return e.clusterClient.Update(cluster)
+		}
+	case "updating":
+		failureMessage, _ := status["failureMessage"].(string)
+
+		if !apimgmtv3.ClusterConditionProvisioned.IsTrue(cluster) {
+			// all provisioning has fnished
+			cluster = cluster.DeepCopy()
+			apimgmtv3.ClusterConditionProvisioned.True(cluster)
+			cluster, err = e.clusterClient.Update(cluster)
+			if err != nil {
+				return cluster, err
+			}
+		}
+
+		if failureMessage != apimgmtv3.ClusterConditionUpdated.GetMessage(cluster) {
+			cluster = cluster.DeepCopy()
+			apimgmtv3.ClusterConditionUpdated.Message(cluster, failureMessage)
+			if failureMessage != "" {
+				apimgmtv3.ClusterConditionUpdated.False(cluster)
+			}
+			cluster, err = e.clusterClient.Update(cluster)
+			if err != nil {
+				return cluster, fmt.Errorf("failure setting updating condition for cluster [%s]: %v", cluster.Name, err)
+			}
+		}
+
+		if !apimgmtv3.ClusterConditionUpdated.IsUnknown(cluster) && !apimgmtv3.ClusterConditionUpdated.IsFalse(cluster) {
+			cluster = cluster.DeepCopy()
+			apimgmtv3.ClusterConditionUpdated.Unknown(cluster)
+			apimgmtv3.ClusterConditionUpdated.Message(cluster, "")
+			cluster, err = e.clusterClient.Update(cluster)
+			if err != nil {
+				return cluster, fmt.Errorf("failure setting updating condition for cluster [%s] to unknown: %v", cluster.Name, err)
+			}
+		}
+
+		if failureMessage == "" {
+			logrus.Infof("waiting for cluster EKS [%s] to update", cluster.Name)
+		} else {
+			logrus.Infof("waiting for cluster EKS [%s] update failure to be resolved", cluster.Name)
+		}
+
+		e.clusterEngqueueAfter(cluster.Name, 10*time.Second)
```

Comment:
Typo `clusterEngqueueAfter`, should try to do a test where this actually gets called to ensure it works as expected.

Reply:
that the EnqueueAfter function works? Neither controller would work if it didn't. When running the controller the logs can be observed on this time interval.